Checkpoint1/Checkpoint1.cs

```diff
@@ -6,7 +6,102 @@ class Program
+            //This one gave me some trouble, had to comb back through the lessons to find break 
+            //which was extremely helpful
+            
+            var sum = 0;
+            while(true){
+                Console.WriteLine("Please enter a number or 'ok' to exit");
+                string numberInput = Console.ReadLine().ToLower();
+                
+
+                if( numberInput == "ok" ){
+                    break;
+                    
+                }
+                sum += Convert.ToInt32(numberInput);
+
+            }
+                    Console.WriteLine("The sum of your numbers is " + sum );
+
+            Console.WriteLine("Press enter to start program 3!");    
+            Console.ReadLine(); 
+            
+      
+            Console.WriteLine("~~~~~~~~~~~~~~~~~~Program 3~~~~~~~~~~~~~~~~~~");
+            //through studying it looks like there is a factorial method somewhere in C#.
+            Console.WriteLine("Give me a number and I will give you it's factorial!");
+            string userInput = Console.ReadLine();
+            int userInt = Convert.ToInt32(userInput);
+            int factorial = userInt;
+            for(int i = userInt -1; i >= 1; i--){
+                    factorial = factorial * i;
+            }
+
+            Console.WriteLine(userInt + "! = " + factorial);
+
+
+            Console.WriteLine("Press enter to start program 4!");    
+            Console.ReadLine();
+
+            Console.WriteLine("~~~~~~~~~~~~~~~~~~Program 4~~~~~~~~~~~~~~~~~~");  
+
+            Random randomInt = new Random();
+            int randomNumber = randomInt.Next(1,10);
+            int winningNumber = randomNumber;
+            
+            //I originally had a line of code to print out the random number
+            //to test win conditions.
+            for(int i = 0; i < 4; i++){
+            Console.WriteLine("Guess a number 1-10!");
+            int userNumber = int.Parse(Console.ReadLine());
+            if( winningNumber == userNumber){
+                Console.WriteLine("You guessed right, congratulations!");
+                i=4;
+            }
+            else if( i >= 3){
+                Console.WriteLine("The number we were looking for was " + winningNumber + ". Sorry, you lose!");
+            }
+            }
+            Console.WriteLine("Press enter to start program 5!");    
+            Console.ReadLine();
+            
+            Console.WriteLine("~~~~~~~~~~~~~~~~~~Program 5~~~~~~~~~~~~~~~~~~");  
+            //Was having an issue with this one where it works sometimes 
+            //than other times it will throw an out of bounds exception
+            Console.WriteLine("Please enter a series of numbers separated by a comma (,).");
+            string userString = Console.ReadLine();
+            string [] noComma = userString.Split(',');
+            int [] numberArray = Array.ConvertAll(noComma, int.Parse);
+            int currentNumber = 0;
```

Comment:
this assumes the user will enter at least 1 positive number. what if they entered all negative numbers?